mkdocs_blogging_plugin/plugin.py

```diff
@@ -258,8 +289,7 @@ def on_page_content(self, html, page, config, files):
         been generated, the time when the meta from markdown file
         has already been added into the page instance.
         """
-
-        if "exclude_from_blog" in page.meta and page.meta["exclude_from_blog"]:
+        if ("exclude_from_blog" in page.meta and page.meta["exclude_from_blog"]) or (self.exclude_index and page.file.name == "index"):
```

Comment:
The condition for excluding the index page should be `self.exclude_index and <this page contains {{ blog_content }}>`. It is not necessary for the index page to be named `index.md`.

Reply:
It's time consuming to check whether the page contains that pattern, so it might be unrealistic to check that here. However, if we do check it here and then cache the result, we can reuse that when we are substituting the pattern in the next step, which keeps the same performance.

Reply:
The index page can have names different than `index.md`, so this condition is not always correct to judge whether the page is an index page. As the correct judgement involves some time-consuming work, i.e. regex matching, it might be not proper to do it here. So I suggest abandon the `exclude_index` setting.